File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class WordCount {
    public Map<String, Integer> phrase(String sentence) {
        final String[] splittedSentence = sentence.replaceAll("[^a-zA-Z0-9 ]", "").split(" ");
        return Stream.of(splittedSentence)
                .filter(w -> !w.isEmpty())
                .map(String::toLowerCase)
                .collect(Collectors.groupingBy(
                        Function.identity(),
                        Collectors.reducing(0, e -> 1, Integer::sum)
                ));
    }
}

File: kotlin/word-count/src/main/kotlin/WordCount.kt
/**
 * Created by yannickgrenzinger on 24/07/2016.
 */
object WordCount {

    fun phrase(sentence: String): Map<String, Int> {
        return sentence.replace(Regex("[^A-Za-z0-9\' ]"),"")
                .toLowerCase()
                .split(" ")
                .filter { w -> w.isNotBlank() }
                .groupBy { it }
                .mapValues { e -> e.value.count() }
    }
    
}